<launch>
    <!-- launching robot Description and gazebo -->
    <include file="$(find rover_description)/launch/rover_description.launch">
        <arg name="world_file" value="worlds/moon_surface.world"/>
        <!--world should be saved in "usr/share/gazebo-7/worlds"-->
        <!-- <arg name="world_file" value="worlds/cube.world"/> -->
        <arg name="gui_state" value="false"/>
    </include>

    <!-- Mapping launch file -->
    <include file="$(find rover_nav)/launch/octo_mapping.launch"/>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find rover_nav)/config/move_base/nav_3d_costmap_params.yaml" command="load" ns="global_costmap"/>
        <rosparam file="$(find rover_nav)/config/move_base/nav_3d_costmap_params.yaml" command="load" ns="local_costmap"/>
        <rosparam file="$(find rover_nav)/config/move_base/base_local_planner_params.yaml" command="load"/>
        <param name="base_global_planner" value="planner_3d/Planner3D"/>
        <param name="controller_frequency" value="20.0"/>
        <param name="planner_frequency" value="10.0"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rover_nav)/rviz/octomap_sim.rviz"/>
</launch>